apis/v1alpha1/route_types.go

```diff
@@ -35,6 +35,26 @@ const (
 	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
 )
 
+const (
+	// AnnotationAppProtocol defines the protocol a Gateway should use for
+	// communication with a Kubernetes Service.
+	// This annotation must be applied to the `BackendPolicy` resource and the
+	// protocol will apply to all Service ports that are selected by
+	// `BackendPolicy.Spec.BackendRefs`. The annotation serves as a stop-gap
+	// solution for the Kubernetes versions < 1.18, where the `AppProtocol`
+	// field on the `Service` resource is not available. Using this annotation
+	// is highly discouraged when AppProtocol field is available. The AppProtocol
+	// field, when populated, takes precedence over this annotation.
+	// The value of the annotation must be validated same as `ProtocolType`
+	// in this package. If invalid, the annotation is ignored.
+	//
+	// Examples:
+	//
+	// - `networking.x-k8s.io/app-protocol: HTTPS`
+	// - `networking.x-k8s.io/app-protocol: TLS`
```

Comment:
I know what I'd said here earlier was rather confusing, but I think it actually makes the most sense to match validation on the AppProtocol field, not our protocol field on Gateways. The key part of that validation is that the value must pass [isQualifiedName](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L42). That's certainly more flexible than our `ProtocolType` validation here. I think that's important for something like AppProtocol. So maybe we should say something like this:

```
The value of this annotation must be also be a valid value for the AppProtocol field.
```